src/tree/FunctionAppTreeItem.ts

```diff
@@ -6,38 +6,65 @@
 import WebSiteManagementClient = require('azure-arm-website');
 import { Site } from 'azure-arm-website/lib/models';
-import * as path from 'path';
 import { OutputChannel } from 'vscode';
-import { SiteWrapper } from 'vscode-azureappservice';
-import { IAzureNode, IAzureTreeItem } from 'vscode-azureextensionui';
+import { AppSettingsTreeItem, AppSettingTreeItem, SiteWrapper } from 'vscode-azureappservice';
+import { IAzureNode, IAzureParentTreeItem, IAzureTreeItem } from 'vscode-azureextensionui';
 import { ArgumentError } from '../errors';
 import { nodeUtils } from '../utils/nodeUtils';
+import { FunctionsTreeItem } from './FunctionsTreeItem';
+import { FunctionTreeItem } from './FunctionTreeItem';
 
-export class FunctionAppTreeItem implements IAzureTreeItem {
-    public static readonly contextValue: string = 'azureFunctionsFunctionApp';
+export class FunctionAppTreeItem implements IAzureParentTreeItem {
+    public static contextValue: string = 'azFuncFunctionApp';
     public readonly contextValue: string = FunctionAppTreeItem.contextValue;
-    public readonly id: string;
     public readonly siteWrapper: SiteWrapper;
 
+    private readonly _functionsTreeItem: FunctionsTreeItem;
+    private readonly _appSettingsTreeItem: AppSettingsTreeItem;
     private readonly _state: string | undefined;
     private readonly _outputChannel: OutputChannel;
 
     public constructor(site: Site, outputChannel: OutputChannel) {
         this.siteWrapper = new SiteWrapper(site);
-        if (!site.id || !site.state) {
+        if (!site.state) {
             throw new ArgumentError(site);
         }
         this._state = site.state;
-        this.id = site.id;
         this._outputChannel = outputChannel;
+        this._functionsTreeItem = new FunctionsTreeItem(this.siteWrapper);
+        this._appSettingsTreeItem = new AppSettingsTreeItem(this.siteWrapper);
+    }
+
+    public get id(): string {
+        return this.siteWrapper.id;
     }
 
     public get label(): string {
         return this._state === 'Running' ? this.siteWrapper.name : `${this.siteWrapper.name} (${this._state})`;
     }
 
-    get iconPath(): string {
-        return path.join(__filename, '..', '..', '..', '..', 'resources', `${FunctionAppTreeItem.contextValue}.svg`);
+    public get iconPath(): string {
+        return nodeUtils.getIconPath(FunctionAppTreeItem.contextValue);
+    }
+
+    public hasMoreChildren(): boolean {
+        return false;
+    }
+
+    public async loadMoreChildren(_node: IAzureNode<IAzureTreeItem>, _clearCache: boolean | undefined): Promise<IAzureTreeItem[]> {
+        return [this._functionsTreeItem, this._appSettingsTreeItem];
+    }
+
+    public pickTreeItem(expectedContextValue: string): IAzureParentTreeItem | undefined {
+        switch (expectedContextValue) {
+            case FunctionTreeItem.contextValue:
+                return this._functionsTreeItem;
+            case AppSettingsTreeItem.contextValue:
```

Comment:
Aren't these different TreeItems?  Should it fall through like that?

Reply:
The node picker is traveling down the tree. In this example, I'm basically saying "If you're looking for a AppSettingsTreeItem _or_ AppSettingTreeItem, go to this. _appSettingsTreeItem next". After that, the node picker will continue down the tree as applicable